pyproject.toml

```diff
@@ -16,11 +16,13 @@ accelerate = "^0.23.0"
 datasets = "^2.14.5"
 scipy = "^1.11.3"
 wandb = "^0.15.12"
+httpx = "^0.26.0"
```

Comment:
seems like we don't need this anymore? Based on your comment, I think this was needed for the e2e test.

For e2e, I wonder if we can do a `modal deply runner` to a `e2e` workspace, tail the modal URL, then call it with our JS scripts? 

Also we should prob add a js script to test auth as well :-?

Reply:
ah good catch! I think I just accidentally forgot to put this in the dev deps group.  IIRC it was required by the fastapi test client, and shortcircuited the unit test before I had added it

--- 

I'll try digging into the E2E + js stuff you mentioned tomorrow :)

Reply:
started the js test by simply putting a bad api key request inside the existing `test-simple.ts` (375823e)

but for the sake of separation & clutter, idk if it makes sense to take it much further inside this PR -- perhaps a new dedicated issue is warranted, to convert these manual scripts to a test suite with assertions?

---

i'm about to get on a flight, so i'll have some time to try out my own deployment & run the manual scripts to ensure what i've pushed so far works.  after that i should have a better sense of how an automated CI/CD for E2E testing could look

Reply:
@louisgv aight yeh the test works as expected now, gonna re-req review

i was able to deploy & do a `pnpm x scripts/test-simple.ts` that resulted in two successful requests & then hit the expected failure of `{"detail":"Incorrect bearer token"}`